onnxruntime/core/providers/nnapi/nnapi_builtin/builders/op_builder.cc

```diff
@@ -400,7 +416,13 @@ static Status AddInitializerTransposed(ModelBuilder& model_builder,
   for (uint32_t x = 0; x < x_t; x++) {
     for (uint32_t y = 0; y < y_t; y++) {
       for (size_t i = 0; i < element_size; i++) {
-        buffer[element_size * (y * x_t + x) + i] = src[element_size * (x * y_t + y) + i];
+        if (is_per_tensor_u8s8) {
+          // Convert int8 tensor to uint8 tensor for per-tensor u8s8
+          // i^0x80 == i + 128
+          buffer[element_size * (y * x_t + x) + i] = src[element_size * (x * y_t + y) + i] ^ 0x80;
```

Comment:
if you hoist the is_per_tensor_u8s8 out of the loops and set a "bit_flip_value" to 0x80 or 0x00 based on the flag, you can simplify the inner loop. This is what we do in MLAS.